src/vsphere_cpi/lib/cloud/vsphere/attach_tag_to_vm.rb

```diff
@@ -0,0 +1,172 @@
+require 'cloud/vsphere/logger'
+require 'vsphere-automation-cis'
+require 'vsphere-automation-vcenter'
+
+module  VSphereCloud
+  module TaggingTag
+    class AttachTagToVm
+      include Logger
+
+      def self.InitializeConnection(cloud_config, logger)
+        configuration = VSphereAutomation::Configuration.new.tap do |config|
+          config.host = cloud_config.vcenter_host
+          config.username = cloud_config.vcenter_user
+          config.password = cloud_config.vcenter_password
+          config.scheme = 'https'
+          config.verify_ssl = false
+          config.verify_ssl_host = nil
+          config.logger = logger
+          config.debugging = false
+        end
+        api_client = VSphereAutomation::ApiClient.new(configuration)
+        api_client.default_headers['Authorization'] = configuration.basic_auth_token
+        session_api = VSphereAutomation::CIS::SessionApi.new(api_client)
+        session_id = session_api.create('').value
+        api_client.default_headers['vmware-api-session-id'] = session_id
+        api_client
+      end
+
+      def initialize(api_client)
+        @api_client = api_client
+      end
+
+      private def tag_assoc_api
```

Comment:
It's not very idiomatic in Ruby to use abbreviations like this. This can just be `tag_association_api`.

Reply:
Updated.